lib/connection-parameters.js

```diff
@@ -43,12 +42,14 @@ var useSsl = function () {
 
 var ConnectionParameters = function (config) {
   // if a string is passed, it is a raw connection string so we parse it into a config
-  config = typeof config === 'string' ? parse(config) : (config || {})
+  config = typeof config === 'string' ? parse(config) : config || {}
+
   // if the config has a connectionString defined, parse IT into the config we use
   // this will override other default values with what is stored in connectionString
-  if (config.connectionString) {
-    config = parse(config.connectionString)
+  if (typeof config.connectionString === 'string') {
```

Comment:
Throwing instead of hiding potential errors is generally good, and it already behaved that way. I’d keep this check unmodified.